src/imitation/algorithms/preference_comparisons.py

```diff
@@ -906,6 +908,158 @@ def _reward_sums(self, fragment_pairs) -> Tuple[np.ndarray, np.ndarray]:
+            if self._display_videos(frag1, frag2):
+                preferences[i] = 1
+        return preferences
+
+    def _display_videos(
+        self, frag1: TrajectoryWithRew, frag2: TrajectoryWithRew
+    ) -> bool:
+        """Displays the videos of the two fragments.
+
+        Args:
+            frag1: first fragment
+            frag2: second fragment
+        """
+        # display the videos
+        frag1_video_path = frag1.infos[0]["video_path"]
+        frag2_video_path = frag2.infos[0]["video_path"]
+        if self._in_ipython():
+            self._display_videos_in_notebook(frag1_video_path, frag2_video_path)
+
+            pref = input(
+                "Which video is preferred? (1 or 2, or q to quit, or r to replay): "
+            )
+            while pref not in ["1", "2", "q"]:
+                if pref == "r":
+                    self._display_videos_in_notebook(frag1_video_path, frag2_video_path)
+                pref = input("Please enter 1 or 2 or q or r: ")
+
+            if pref == "q":
+                raise KeyboardInterrupt
+            elif pref == "1":
+                return True
+            elif pref == "2":
+                return False
+
+            # should never be hit
+            raise ValueError(f"Unexpected input {pref}")
+        else:
+            print("Which video is preferred? (1 or 2, or q to quit, or r to replay):\n")
+            cap1 = cv2.VideoCapture(str(frag1_video_path))
+            cap2 = cv2.VideoCapture(str(frag2_video_path))
+            cv2.namedWindow("Video 1", cv2.WINDOW_NORMAL)
+            cv2.namedWindow("Video 2", cv2.WINDOW_NORMAL)
+
+            # set window sizes
+            cv2.resizeWindow("Video 1", 500, 500)
+            cv2.resizeWindow("Video 2", 500, 500)
+
+            # move windows side by side
+            cv2.moveWindow("Video 1", 0, 0)
+            cv2.moveWindow("Video 2", 500, 0)
+
+            if not cap1.isOpened():
+                raise RuntimeError(f"Error opening video file {frag1_video_path}.")
+
+            if not cap2.isOpened():
+                raise RuntimeError(f"Error opening video file {frag2_video_path}.")
+
+            ret1, frame1 = cap1.read()
+            ret2, frame2 = cap2.read()
+            while cap1.isOpened() and cap2.isOpened():
+                if ret1 or ret2:
+                    cv2.imshow("Video 1", frame1)
+                    cv2.imshow("Video 2", frame2)
+                    ret1, frame1 = cap1.read()
+                    ret2, frame2 = cap2.read()
+
+                key = chr(cv2.waitKey(1) & 0xFF)
+                if key == "q":
+                    raise KeyboardInterrupt
+                elif key == "r":
+                    cap1.set(cv2.CAP_PROP_POS_FRAMES, 0)
+                    cap2.set(cv2.CAP_PROP_POS_FRAMES, 0)
+                    ret1, frame1 = cap1.read()
+                    ret2, frame2 = cap2.read()
+                elif key == "1" or key == "2":
+                    cap1.release()
```

Comment:
Do we always want these lines to be executed? If so, might want to move them to a `finally:` block to both ensure they always execute and cut duplication of lines 1026-1028.

Reply:
Maybe I'm wrong but based on Googling it looks like `finally` can only be used for try/except blocks? Were you thinking I'd add a dummy try/except here?

Reply:
Or for the KeyboardInterrupt case? TBH I think I don't need to run these functions in the KeyboardInterrupt case because it's going to kill the program anyway